Jaeger Opentelemetry Ingester cannot correctly handle `kafka.consumer.encoding` flag

**Describe the bug**
Jaeger Opentelemetry Ingester cannot correctly handle `kafka.consumer.encoding` when its value is `zipkin-thrift`. Also, the flag not accept the `otlp-proto` as value even Ingester's Kafka implementation is from OTel.

**To Reproduce**
1. Check helper
```bash
$ opentelemetry-ingester -h

...
--kafka.consumer.encoding string                     The encoding of spans ("json", "protobuf", "zipkin-thrift") consumed from kafka (default "protobuf")
...
```

2. Use `zipkin-thrift` as encoding value

```bash
$ opentelemetry-ingester --kafka.consumer.encoding zipkin-thrift
2020-10-20T14:04:31.624+0800	INFO	service/service.go:396	Starting Jaeger OpenTelemetry Ingester...	{"Version": "v1.20.0", "GitHash": "bd59f13478cdd35b41e85f67a6f04b6cb4d92df8", "NumCPU": 4}
2020-10-20T14:04:31.625+0800	INFO	service/service.go:240	Setting up own telemetry...
2020-10-20T14:04:31.629+0800	INFO	service/telemetry.go:110	Serving Prometheus metrics	{"address": "localhost:8888", "legacy_metrics": false, "new_metrics": true, "level": 0, "service.instance.id": "28b6a4ce-d467-4a53-8941-06840458019e"}
panic: zipkin-thrift is not a supported kafka encoding in the OTEL collector.

goroutine 1 [running]:
github.com/jaegertracing/jaeger/cmd/opentelemetry/app/receiver/kafkareceiver.MustOtelEncodingForJaegerEncoding(...)
	github.com/jaegertracing/jaeger/cmd/opentelemetry/app/receiver/kafkareceiver/kafka_receiver.go:158
github.com/jaegertracing/jaeger/cmd/opentelemetry/app/receiver/kafkareceiver.(*Factory).CreateDefaultConfig(0xc000488140, 0x46774a0, 0xc0008ee4b0)
	github.com/jaegertracing/jaeger/cmd/opentelemetry/app/receiver/kafkareceiver/kafka_receiver.go:62 +0x7e8
go.opentelemetry.io/collector/config/configcheck.ValidateConfigFromFactories(0xc0003de840, 0xc0003de900, 0xc0003de8d0, 0xc0003de7e0, 0x0, 0xc000407560)
	go.opentelemetry.io/collector@v0.12.0/config/configcheck/configcheck.go:41 +0xe2
go.opentelemetry.io/collector/service.(*Application).setupConfigurationComponents(0xc000846000, 0x468dba0, 0xc0000560c8, 0xc000409610, 0x0, 0x3)
	go.opentelemetry.io/collector@v0.12.0/service/service.go:273 +0x5a
go.opentelemetry.io/collector/service.(*Application).execute(0xc000846000, 0x468dba0, 0xc0000560c8, 0xc000409610, 0x0, 0x0)
	go.opentelemetry.io/collector@v0.12.0/service/service.go:414 +0x4e5
go.opentelemetry.io/collector/service.New.func1(0xc00004e580, 0xc000489780, 0x0, 0x2, 0x0, 0x0)
	go.opentelemetry.io/collector@v0.12.0/service/service.go:154 +0xa5
github.com/spf13/cobra.(*Command).execute(0xc00004e580, 0xc00004c1c0, 0x2, 0x2, 0xc00004e580, 0xc00004c1c0)
	github.com/spf13/cobra@v1.0.0/command.go:842 +0x47c
github.com/spf13/cobra.(*Command).ExecuteC(0xc00004e580, 0x0, 0x0, 0x3)
	github.com/spf13/cobra@v1.0.0/command.go:950 +0x375
github.com/spf13/cobra.(*Command).Execute(...)
	github.com/spf13/cobra@v1.0.0/command.go:887
go.opentelemetry.io/collector/service.(*Application).Run(...)
	go.opentelemetry.io/collector@v0.12.0/service/service.go:467
main.main()
	command-line-arguments/main.go:97 +0x8af
```

**Expected behavior**
Accept `zipkin-thrift` or `otlp-proto`as its value since these values (nearly) work in the config file.

**Version (please complete the following information):**
 - OS: macOS 10.15.6
 - Jaeger version: 3d998711700382c6729b83fa66f8b20a3aae5116 (master)

--- a/cmd/opentelemetry/app/exporter/badgerexporter/doc.go
+++ b/cmd/opentelemetry/app/exporter/badgerexporter/doc.go
@@ -12,6 +12,5 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-
 // Package badgerexporter implements Jaeger Badger storage as OpenTelemetry exporter.
 package badgerexporter

--- a/cmd/opentelemetry/app/exporter/kafkaexporter/flags.go
+++ b/cmd/opentelemetry/app/exporter/kafkaexporter/flags.go
@@ -16,12 +16,25 @@ package kafkaexporter
 
 import (
 	"flag"
+	"fmt"
 
 	"github.com/jaegertracing/jaeger/plugin/storage/kafka"
+)
+
+const (
+	// encodingOTLPProto is used for spans encoded as OTLP Protobuf.
+	encodingOTLPProto = "otlp-proto"
 )
 
 // AddFlags adds Ingester flags.
 func AddFlags(flags *flag.FlagSet) {
 	opts := &kafka.Options{}
 	opts.AddOTELFlags(flags)
+	// Modify kafka.producer.encoding flag
+	flags.Lookup("kafka.producer.encoding").Usage = fmt.Sprintf(
+		`Encoding of spans ("%s", "%s" or "%s") sent to kafka.`,
+		kafka.EncodingJSON,
+		kafka.EncodingProto,
+		encodingOTLPProto,
+	)
 }

--- a/cmd/opentelemetry/app/exporter/kafkaexporter/kafka_exporter.go
+++ b/cmd/opentelemetry/app/exporter/kafkaexporter/kafka_exporter.go
@@ -134,6 +134,8 @@ func mustOtelEncodingForJaegerEncoding(jaegerEncoding string) string {
 		return "jaeger_proto"
 	case kafka.EncodingJSON:
 		return "jaeger_json"
+	case encodingOTLPProto:
+		return "otlp_proto"
 	}
 
 	panic(jaegerEncoding + " is not a supported kafka encoding in the OTEL collector.")

--- a/cmd/opentelemetry/app/exporter/kafkaexporter/kafka_exporter.go
+++ b/cmd/opentelemetry/app/exporter/kafkaexporter/kafka_exporter.go
@@ -23,7 +23,6 @@ import (
 	"go.opentelemetry.io/collector/config/configtls"
 	"go.opentelemetry.io/collector/exporter/kafkaexporter"
 
-	"github.com/jaegertracing/jaeger/cmd/opentelemetry/app/receiver/kafkareceiver"
 	"github.com/jaegertracing/jaeger/plugin/storage/kafka"
 )
 
@@ -59,7 +58,7 @@ func (f Factory) CreateDefaultConfig() configmodels.Exporter {
 	opts := &kafka.Options{}
 	opts.InitFromViper(f.Viper)
 
-	cfg.Encoding = kafkareceiver.MustOtelEncodingForJaegerEncoding(opts.Encoding)
+	cfg.Encoding = mustOtelEncodingForJaegerEncoding(opts.Encoding)
 	cfg.Topic = opts.Topic
 	cfg.Brokers = opts.Config.Brokers
 	cfg.ProtocolVersion = opts.Config.ProtocolVersion
@@ -126,4 +125,16 @@ func (f Factory) CreateLogsExporter(
 	cfg configmodels.Exporter,
 ) (component.LogsExporter, error) {
 	return f.Wrapped.CreateLogsExporter(ctx, params, cfg)
+}
+
+// mustOtelEncodingForJaegerEncoding translates a jaeger encoding to a otel encoding
+func mustOtelEncodingForJaegerEncoding(jaegerEncoding string) string {
+	switch jaegerEncoding {
+	case kafka.EncodingProto:
+		return "jaeger_proto"
+	case kafka.EncodingJSON:
+		return "jaeger_json"
+	}
+
+	panic(jaegerEncoding + " is not a supported kafka encoding in the OTEL collector.")
 }

--- a/cmd/opentelemetry/app/exporter/kafkaexporter/kafka_exporter_test.go
+++ b/cmd/opentelemetry/app/exporter/kafkaexporter/kafka_exporter_test.go
@@ -27,6 +27,7 @@ import (
 
 	"github.com/jaegertracing/jaeger/cmd/flags"
 	jConfig "github.com/jaegertracing/jaeger/pkg/config"
+	"github.com/jaegertracing/jaeger/plugin/storage/kafka"
 )
 
 func TestDefaultConfig(t *testing.T) {
@@ -85,4 +86,34 @@ func TestLoadConfigAndFlags(t *testing.T) {
 	assert.Equal(t, "jaeger", kafkaCfg.Authentication.Kerberos.Realm)
 	assert.Equal(t, "/etc/foo", kafkaCfg.Authentication.Kerberos.ConfigPath)
 	assert.Equal(t, "from-jaeger-config", kafkaCfg.Authentication.Kerberos.Username)
+}
+
+func TestMustOtelEncodingForJaegerEncoding(t *testing.T) {
+	tests := []struct {
+		in           string
+		expected     string
+		expectsPanic bool
+	}{
+		{
+			in:       kafka.EncodingProto,
+			expected: "jaeger_proto",
+		},
+		{
+			in:       kafka.EncodingJSON,
+			expected: "jaeger_json",
+		},
+		{
+			in:           "not-an-encoding",
+			expectsPanic: true,
+		},
+	}
+
+	for _, tt := range tests {
+		if tt.expectsPanic {
+			assert.Panics(t, func() { mustOtelEncodingForJaegerEncoding(tt.in) })
+			continue
+		}
+
+		assert.Equal(t, tt.expected, mustOtelEncodingForJaegerEncoding(tt.in))
+	}
 }

--- a/cmd/opentelemetry/app/exporter/kafkaexporter/kafka_exporter_test.go
+++ b/cmd/opentelemetry/app/exporter/kafkaexporter/kafka_exporter_test.go
@@ -103,6 +103,10 @@ func TestMustOtelEncodingForJaegerEncoding(t *testing.T) {
 			expected: "jaeger_json",
 		},
 		{
+			in:       encodingOTLPProto,
+			expected: "otlp_proto",
+		},
+		{
 			in:           "not-an-encoding",
 			expectsPanic: true,
 		},

--- a/cmd/opentelemetry/app/receiver/kafkareceiver/flags.go
+++ b/cmd/opentelemetry/app/receiver/kafkareceiver/flags.go
@@ -16,11 +16,30 @@ package kafkareceiver
 
 import (
 	"flag"
+	"fmt"
+	"strings"
 
 	ingesterApp "github.com/jaegertracing/jaeger/cmd/ingester/app"
+	"github.com/jaegertracing/jaeger/plugin/storage/kafka"
+)
+
+const (
+	// encodingZipkinProto is used for spans encoded as Zipkin Protobuf.
+	encodingZipkinProto = "zipkin-proto"
+	// encodingZipkinJSON is used for spans encoded as Zipkin JSON.
+	encodingZipkinJSON = "zipkin-json"
+	// encodingOTLPProto is used for spans encoded as OTLP Protobuf.
+	encodingOTLPProto = "otlp-proto"
 )
 
 // AddFlags adds Ingester flags.
 func AddFlags(flags *flag.FlagSet) {
 	ingesterApp.AddOTELFlags(flags)
+	// Modify kafka.consumer.encoding flag
+	flags.Lookup(ingesterApp.KafkaConsumerConfigPrefix + ingesterApp.SuffixEncoding).Usage = fmt.Sprintf(`The encoding of spans ("%s") consumed from kafka`, strings.Join(
+		append(kafka.AllEncodings,
+			encodingZipkinJSON,
+			encodingZipkinProto,
+			encodingOTLPProto,
+		), "\", \""))
 }

--- a/cmd/opentelemetry/app/receiver/kafkareceiver/kafka_receiver.go
+++ b/cmd/opentelemetry/app/receiver/kafkareceiver/kafka_receiver.go
@@ -59,7 +59,7 @@ func (f *Factory) CreateDefaultConfig() configmodels.Receiver {
 
 	cfg.Brokers = opts.Brokers
 	cfg.ClientID = opts.ClientID
-	cfg.Encoding = MustOtelEncodingForJaegerEncoding(opts.Encoding)
+	cfg.Encoding = mustOtelEncodingForJaegerEncoding(opts.Encoding)
 	cfg.GroupID = opts.GroupID
 	cfg.Topic = opts.Topic
 	cfg.ProtocolVersion = opts.ProtocolVersion
@@ -138,13 +138,21 @@ func (f Factory) CreateLogsReceiver(
 	return f.Wrapped.CreateLogsReceiver(ctx, params, cfg, nextConsumer)
 }
 
-// MustOtelEncodingForJaegerEncoding translates a jaeger encoding to a otel encoding
-func MustOtelEncodingForJaegerEncoding(jaegerEncoding string) string {
+// mustOtelEncodingForJaegerEncoding translates a jaeger encoding to a otel encoding
+func mustOtelEncodingForJaegerEncoding(jaegerEncoding string) string {
 	switch jaegerEncoding {
 	case kafka.EncodingProto:
 		return "jaeger_proto"
 	case kafka.EncodingJSON:
 		return "jaeger_json"
+	case encodingOTLPProto:
+		return "otlp_proto"
+	case encodingZipkinProto:
+		return "zipkin_proto"
+	case encodingZipkinJSON:
+		return "zipkin_json"
+	case kafka.EncodingZipkinThrift:
+		return "zipkin_thrift"
 	}
 
 	panic(jaegerEncoding + " is not a supported kafka encoding in the OTEL collector.")

--- a/cmd/opentelemetry/app/receiver/kafkareceiver/kafka_receiver_test.go
+++ b/cmd/opentelemetry/app/receiver/kafkareceiver/kafka_receiver_test.go
@@ -105,6 +105,22 @@ func TestMustOtelEncodingForJaegerEncoding(t *testing.T) {
 			expected: "jaeger_json",
 		},
 		{
+			in:       encodingOTLPProto,
+			expected: "otlp_proto",
+		},
+		{
+			in:       encodingZipkinProto,
+			expected: "zipkin_proto",
+		},
+		{
+			in:       encodingZipkinJSON,
+			expected: "zipkin_json",
+		},
+		{
+			in:       kafka.EncodingZipkinThrift,
+			expected: "zipkin_thrift",
+		},
+		{
 			in:           "not-an-encoding",
 			expectsPanic: true,
 		},
@@ -112,19 +128,10 @@ func TestMustOtelEncodingForJaegerEncoding(t *testing.T) {
 
 	for _, tt := range tests {
 		if tt.expectsPanic {
-			assertPanic(t, func() { MustOtelEncodingForJaegerEncoding(tt.in) })
+			assert.Panics(t, func() { mustOtelEncodingForJaegerEncoding(tt.in) })
 			continue
 		}
 
-		assert.Equal(t, tt.expected, MustOtelEncodingForJaegerEncoding(tt.in))
+		assert.Equal(t, tt.expected, mustOtelEncodingForJaegerEncoding(tt.in))
 	}
-}
-
-func assertPanic(t *testing.T, f func()) {
-	defer func() {
-		if r := recover(); r == nil {
-			t.Errorf("The code did not panic")
-		}
-	}()
-	f()
 }

